llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/LlapTaskSchedulerService.java

```diff
@@ -1447,8 +1447,14 @@ private SelectHostResult selectHost(TaskInfo request, Map<String, List<NodeInfo>
                 if (request.shouldForceLocality()) {
                   requestedHostsWillBecomeAvailable = true;
                 } else {
-                  LlapServiceInstance inst = activeInstances.getByHost(host).stream().findFirst().get();
-                  NodeInfo nodeInfo = instanceToNodeMap.get(inst.getWorkerIdentity());
+                  Set<LlapServiceInstance> instanceTypes = activeInstances.getByHost(host);
```

Comment:
Looks like this may happen when a node goes down between getResourceAvailability() call until selectHost() is triggered. Following the previous logic I believe the check should be performed at the same level as: 
```availableHostMap.containsKey(host)``` as these type of requests should not be waiting for requestedHostsWillBecomeAvailable.